client/src/pages/signup.tsx

```diff
@@ -1,142 +1,135 @@
-import axios from "axios";
-import Image from "next/image";
-import { FormEvent, ReactElement, useState } from "react";
+import { useEffect, useState } from "react";
 
-import { Button } from "@/components/ui/button";
-import {
-  InputEmailPhone,
-  InputImage,
-  InputName,
-} from "@/components/ui/signup/multi-step-form";
+import { Ava } from "@/components/ui/signup/avatar";
+import { Bio } from "@/components/ui/signup/bio";
+import { MainPage } from "@/components/ui/signup/main-page";
 import { ProgressBar } from "@/components/ui/signup/progress-bar";
-import { MultistepForm, useMultistepForm } from "@/hooks/use-multistep-form";
+import { PW } from "@/components/ui/signup/pw";
+import { Welcome } from "@/components/ui/signup/welcome";
+import api from "@/lib/api";
 
-interface FormData {
-  email?: string;
-  phone?: string;
-  contactMethod: string;
-  fullname: string;
-  avatarSrc: string;
-}
+const BIDDER = "Bidder";
+const POSTER = "Poster";
 
-const INIT_FORM_DATA: FormData = {
-  email: "",
-  phone: "",
-  contactMethod: "",
-  fullname: "",
-  avatarSrc: "",
-};
-
-const POSTER: string = "Poster";
-const BIDDER: string = "Bidder";
+const SignUp: React.FC = () => {
+  const [client, setClient] = useState("Poster");
+  const [currentStep, setCurrentStep] = useState(1);
+  const [emailMobile, setEmailMobile] = useState("");
+  const [fullName, setFullName] = useState("");
+  const [password, setPassword] = useState("");
+  const [profilePhoto, setProfilePhoto] = useState<string | undefined>(
+    undefined,
+  );
+  const [bio, setBio] = useState("");
+  const [submit, setSubmit] = useState(false);
+  const [title, setTitle] = useState("Welcome!");
 
-function SignUp() {
-  const BackArrow = "/back-arrow.svg";
-  const [user, setUser] = useState(POSTER);
-  const [data, setData] = useState(INIT_FORM_DATA);
-  function updateFields(fields: Partial<FormData>) {
-    setData((curFormData) => {
-      return { ...curFormData, ...fields };
-    });
-  }
+  useEffect(() => {
+    const onFinish = async () => {
+      const data = {
+        email: emailMobile,
+        username: fullName,
+        password: password,
+        //avatar: profilePhoto,
+        //client: Poster or Bidder
+        //Bio: if it's Bidder, record Bio
+      };
 
-  let forms: ReactElement[] = [
-    <InputEmailPhone {...data} updateFields={updateFields} />,
-    <InputName {...data} updateFields={updateFields} />,
-    <InputImage {...data} updateFields={updateFields} />,
-  ];
-  const multistepForm: MultistepForm = useMultistepForm(forms);
+      try {
+        const response = await api.post("/app/register/", data);
```

Comment:
In the sign up page, you should send two post requests. One is sending to "/app/register/", another one is sending to "/app/profiles/". @yunho7687 Is this how it works?
<img width="1101" alt="image" src="https://github.com/user-attachments/assets/38da18b7-16aa-45e2-b6b0-2110badc7e81">

<img width="1325" alt="image" src="https://github.com/user-attachments/assets/26b629b8-c507-4674-ad89-07b59de57d2f">

Reply:
/app/profile was destoryed. According to @yunho7687